### PR TITLE
WT-7784 Enable RTS to use checkpoint snapshot on timestamp tables

### PR DESCRIPTION
**Pull request by kommiharibabu**
Usually for timestamped tables, the commit/durable timestamp will be
always greater than the stable timestamp and there is no need of
checkpoint snapshot to abort unstable updates. Allowing it to use
checkpoint snapshot shouldn't be a problem unless the commit/durable
timestamp doesn't adhere to the condition of that it must be less
than the stable timestamp.

### CODE REVIEW

**Review by quchenhao — COMMENTED** [review]
```
    /*
     * TODO: We are deliberately not using a snapshot when checkpoint is active. This will ensure
     * that point-in-time checkpoints have a consistent version of data. Remove this condition once
     * fuzzy transaction ID based checkpoints work is merged (WT-6673).
     */
```
Can we change this with this change? Have we run a mongodb patch build and stress testing with this change?

**Comment by kommiharibabu** [comment]
@quchenhao I feel it is better to let that change be separate from it. I do remember that there should be another ticket to enable it. Once this change is stabilized and didn't get any problems, that ticket can be scheduled.

Yes, I ran both stress tests and MongoDB patch build and they are successful.

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -8,9 +8,8 @@
 
 #include "wt_internal.h"
 
-#define WT_CHECK_RECOVERY_FLAG_TS_TXNID(session, txnid, durablets)             \
-    ((durablets) == WT_TS_NONE && F_ISSET(S2C(session), WT_CONN_RECOVERING) && \
-      (txnid) >= S2C(session)->recovery_ckpt_snap_min)
+#define WT_CHECK_RECOVERY_FLAG_TS_TXNID(session, txnid) \
```
Comment (raviprakashgiri29) [inline src/txn/txn_rollback_to_stable.c:11]: Can we rename the macro to `WT_CHECK_RECOVERY_FLAG_TXNID` since we are not comparing the timestamp anymore. ?

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM with some a minor comment.

**Review by hbokhari — APPROVED** [review]
LGTM - nit Ravi's comment about renaming `WT_CHECK_RECOVERY_FLAG_TS_TXNID`

File: src/txn/txn_rollback_to_stable.c (hunk 1 above)
Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c]: Fixed.

**Pull request merged**